diff --git a/base/core/java/android/app/ActivityThread.java b/base/core/java/android/app/ActivityThread.java
index 9056287..b4c34bb 100644
--- a/base/core/java/android/app/ActivityThread.java
+++ b/base/core/java/android/app/ActivityThread.java
@@ -104,6 +104,7 @@ import android.view.WindowManager;
 import android.view.WindowManagerGlobal;
 import android.renderscript.RenderScriptCacheDir;
 import android.security.keystore.AndroidKeyStoreProvider;
+import android.security.keystore.SoterKeyStoreProvider;
 
 import com.android.internal.app.IVoiceInteractor;
 import com.android.internal.content.ReferrerIntent;
@@ -5645,6 +5646,8 @@ public final class ActivityThread {
         EventLogger.setReporter(new EventLoggingReporter());
 
         AndroidKeyStoreProvider.install();
+        // Add Soter provider
+        SoterKeyStoreProvider.install();
 
         // Make sure TrustedCertificateStore looks in the right place for CA certificates
         final File configDir = Environment.getUserConfigDirectory(UserHandle.myUserId());
diff --git a/base/core/java/android/security/keymaster/KeymasterDefs.java b/base/core/java/android/security/keymaster/KeymasterDefs.java
index 04d5952..de9ef35 100644
--- a/base/core/java/android/security/keymaster/KeymasterDefs.java
+++ b/base/core/java/android/security/keymaster/KeymasterDefs.java
@@ -85,6 +85,19 @@ public final class KeymasterDefs {
     public static final int KM_TAG_AUTH_TOKEN = KM_BYTES | 1002;
     public static final int KM_TAG_MAC_LENGTH = KM_UINT | 1003;
 
+    /*
+     * Below for soter usage
+     */
+    public static final int KM_TAG_SOTER_IS_FROM_SOTER = KM_BOOL | 11000;
+    public static final int KM_TAG_SOTER_IS_AUTO_SIGNED_WITH_ATTK_WHEN_GET_PUBLIC_KEY = KM_BOOL | 11001;
+    public static final int KM_TAG_SOTER_IS_AUTO_SIGNED_WITH_COMMON_KEY_WHEN_GET_PUBLIC_KEY = KM_BOOL | 11002;
+    public static final int KM_TAG_SOTER_AUTO_SIGNED_COMMON_KEY_WHEN_GET_PUBLIC_KEY = KM_BYTES | 11003;
+    public static final int KM_TAG_SOTER_AUTO_ADD_COUNTER_WHEN_GET_PUBLIC_KEY = KM_BOOL | 11004;
+    public static final int KM_TAG_SOTER_IS_SECMSG_FID_COUNTER_SIGNED_WHEN_SIGN = KM_BOOL | 11005;
+    public static final int KM_TAG_SOTER_USE_NEXT_ATTK = KM_BOOL | 11006;
+    public static final int KM_TAG_SOTER_UID = KM_UINT | 11007;
+    public static final int KM_TAG_SOTER_AUTO_SIGNED_COMMON_KEY_WHEN_GET_PUBLIC_KEY_BLOB = KM_UINT | 11008;
+
     // Algorithm values.
     public static final int KM_ALGORITHM_RSA = 1;
     public static final int KM_ALGORITHM_EC = 3;
