diff --git a/keymaster/keymaster_enforcement.cpp b/keymaster/keymaster_enforcement.cpp
index 4d0aaef..85c5bfa 100644
--- a/keymaster/keymaster_enforcement.cpp
+++ b/keymaster/keymaster_enforcement.cpp
@@ -321,6 +321,15 @@ keymaster_error_t KeymasterEnforcement::AuthorizeBegin(const keymaster_purpose_t
         case KM_TAG_APPLICATION_ID:
         case KM_TAG_USER_ID:
         case KM_TAG_ALL_USERS:
+        case KM_TAG_SOTER_IS_FROM_SOTER:
+        case KM_TAG_SOTER_IS_AUTO_SIGNED_WITH_ATTK_WHEN_GET_PUBLIC_KEY:
+        case KM_TAG_SOTER_IS_AUTO_SIGNED_WITH_COMMON_KEY_WHEN_GET_PUBLIC_KEY:
+        case KM_TAG_SOTER_AUTO_SIGNED_COMMON_KEY_WHEN_GET_PUBLIC_KEY:
+        case KM_TAG_SOTER_AUTO_ADD_COUNTER_WHEN_GET_PUBLIC_KEY:
+        case KM_TAG_SOTER_IS_SECMSG_FID_COUNTER_SIGNED_WHEN_SIGN:
+        case KM_TAG_SOTER_USE_NEXT_ATTK:
+        case KM_TAG_SOTER_UID:
+        case KM_TAG_SOTER_AUTO_SIGNED_COMMON_KEY_WHEN_GET_PUBLIC_KEY_BLOB:
             break;
 
         case KM_TAG_BOOTLOADER_ONLY:
