diff --git a/init.cpp b/init.cpp
index 6c08982..ab49855 100644
--- a/init.cpp
+++ b/init.cpp
@@ -166,7 +166,7 @@ static int wait_for_coldboot_done_action(const std::vector<std::string>& args) {
     // Any longer than 1s is an unreasonable length of time to delay booting.
     // If you're hitting this timeout, check that you didn't make your
     // sepolicy regular expressions too expensive (http://b/19899875).
-    if (wait_for_file(COLDBOOT_DONE, 1)) {
+    if (wait_for_file(COLDBOOT_DONE, 5)) {
         ERROR("Timed out waiting for %s\n", COLDBOOT_DONE);
     }
 
